sympy/printing/tests/test_c.py

```diff
@@ -30,6 +30,15 @@
 x, y, z = symbols('x,y,z')
 
 
+def test_invalid_variable_names():
+    long_var = 'w'*32
+    syms = symbols('I_{x}, a*, 9t, bus#, %s' % long_var)
```

Comment:
I guess if you did `symbols('a[0]') this would not raise and it should. It should raise for a symbols named `a[0]` but not printing an indexed symbol that results in `a[0]`.

Reply:
That would break tons of existing code. I've seen plenty of people (me included) which have used symbol names like `"a[I]"` for code-gen purposes.

Reply:
Yes, it is unintended that `a[xxx]` wouldn't work.

Reply:
Oh, sorry, I misread what you wrote.